WIP feat: introducing GPU services (#3678)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.0/gpu/index.md

---
layout: layout.pug
navigationTitle: GPUs
title: GPUs
menuWeight: 150
excerpt: Configure GPU for Kommander cluster
beta: false
enterprise: false
---

<!-- markdownlint-disable MD004 MD007 MD025 MD030 -->

> This section describes the Nvidia GPU support on Kommander. This document assumes familiarity with Kubernetes GPU support. AMD GPUs are currently not supported.

Reviewer: Why do we specifically name AMD GPUs here? Would it be more general to say `Currently Kommander only supports Nvidia GPUs.`

# Kommander GPU Overview

GPU support on Kommander is achieved by using the [nvidia-container-runtime][nvidia_container_runtime].
With the Nvidia GPU turned on, Kommander configures the container runtime for running GPU containers, and installs all the necessary items to power up the Nvidia GPU devices.

> Kommander runs every Nvidia GPU dependent component as daemonsets, making it easier to manage and upgrade.

Reviewer: Perhaps move this sentence below the list of components. That would also make it more natural to say `Kommander runs these components as daemonsets,...`

| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla K80           Off  | 00000000:00:1E.0 Off |                    0 |
| N/A   35C    P0    73W / 149W |      0MiB / 11441MiB |     99%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

## Configuration

### Enable Nvidia GPU on Kommander

> To enable Nvidia GPU support on Kommander the first thing to do is adding the label `konvoy.mesosphere.com/gpu-provider=NVIDIA` to each GPU node with installed Nvidia host components.

Reviewer: ```suggestion
To enable Nvidia GPU support on Kommander add the label `konvoy.mesosphere.com/gpu-provider=NVIDIA` to each GPU node with installed Nvidia host components.
```

+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

## Configuration

### Enable Nvidia GPU on Kommander

To enable Nvidia GPU support on Kommander the first thing to do is adding the label `konvoy.mesosphere.com/gpu-provider=NVIDIA` to each GPU node with installed Nvidia host components.

```bash
kubectl label nodes <NODE> konvoy.mesosphere.com/gpu-provider=NVIDIA
```

Then when installing Kommander, enable the `nvidia` services. 

```bash
helm install -n kommander --create-namespace kommander-bootstrap kommander/kommander-bootstrap --set services.nvidia.enabled=true

> ```

Author: This is the essential
